src/Datadog.AutoInstrumentation.NativeLoader/dynamic_dispatcher.cpp

```diff
@@ -0,0 +1,332 @@
+            "win-x64", "linux-x64", "osx-x64",
+            "win-x86", "linux-x86", "osx-x86",
+            "win-arm64", "linux-arm64", "osx-arm64",
+            "win-arm", "linux-arm", "osx-arm",
+        };
+
+        while (t)
+        {
+            std::string line;
+            std::getline(t, line);
+            line = Trim(line);
+            if (line.length() != 0)
+            {
+                Debug(line);
+
+                if (line[0] == '#')
+                {
+                    continue;
+                }
+
+                std::vector<std::string> lineArray = Split(line, ';');
+                if (lineArray.size() != 4)
+                {
+                    Warn("DynamicDispatcherImpl::LoadConfiguration: Invalid line: ", line);
+                }
+
+                std::string type = lineArray[0];
+                std::string idValue = lineArray[1];
+                std::string osArchValue = lineArray[2];
+                std::string filepathValue = lineArray[3];
+
+                if (type == "TRACER" || type == "PROFILER" || type == "CUSTOM")
+                {
+                    if (osArchValue == currentOsArch)
+                    {
+                        // Convert possible relative paths to absolute paths using the configuration file folder as base
+                        // (current_path)
+                        std::string absoluteFilepathValue = std::filesystem::absolute(filepathValue).string();
+                        Debug("DynamicDispatcherImpl::LoadConfiguration: [", type, "] Loading: ", filepathValue, " [AbsolutePath=", absoluteFilepathValue,"]");
+                        if (std::filesystem::exists(absoluteFilepathValue))
+                        {
+                            Debug("[", type, "] Creating a new DynamicInstance object");
+
+                            if (type == "TRACER")
+                            {
+                                this->m_tracerInstance =
+                                    std::make_unique<DynamicInstanceImpl>(absoluteFilepathValue, idValue);
+                            }
+                            else if (type == "PROFILER")
+                            {
+                                this->m_continuousProfilerInstance =
+                                    std::make_unique<DynamicInstanceImpl>(absoluteFilepathValue, idValue);
+                            }
+                            else if (type == "CUSTOM")
+                            {
+                                this->m_customInstance =
+                                    std::make_unique<DynamicInstanceImpl>(absoluteFilepathValue, idValue);
+                            }
+
+                            WSTRING env_key = WStr("PROFID_") + ToWSTRING(idValue);
+                            WSTRING env_value = ToWSTRING(absoluteFilepathValue);
+                            Debug("DynamicDispatcherImpl::LoadConfiguration: [", type, "] Setting environment variable: ", env_key, "=", env_value);
```

Comment:
Because the tracer uses the value from `CORECLR_PROFILER_PATH_XX` or `COR_PROFILER_PATH_XX` to set the PInvokes map for non windows platform using the absolute path. And for the tracer those values should contain the Tracer profiler filepath. Now with the native loader this is not the case, so the PInvoke will redirect the calls to the native loader, and failing due unexisting export methods. So we need to enable a way so the tracer can read it's own profiler filepath.

Reply:
See the comment below, the key was changed to `DD_TRACER_PROFILER_PATH` and `DD_CONTINUOUSPROFILER_PROFILER_PATH`

Reply:
Env Vars are seen by the user and are, essentially, public APIs. In profiling, to to avoid the confusion between
a) the Profiler as in the _Continuous Profiler product_ and
b) the Profiler as in the native part of the Tracer product that uses the .NET profiling iface
we use the term Native Profiler Engine for (b).

We also use the `DD_PROFILING_` prefix to be consistent with other languages who already established a convention. Tracer uses a `DD_TRACE_` prefix. We are also encouraged to make it clear about APIs that must be public for technical reasons, but are not intended for public use (`internal` prefix).

So, for consistency with all that, how do you feel about the following names:

`DD_INTERNAL_PROFILING_NATIVE_ENGINE_PATH`
`DD_INTERNAL_TRACE_NATIVE_ENGINE_PATH`
`DD_INTERNAL_CUSTOM_CLR_PROFILER_PATH` or `DD_INTERNAL_CUSTOM_NATIVE_ENGINE_PATH`
?

Reply:
❗ [follow up in future PR]

Reply:
Changed to `DD_INTERNAL_PROFILING_NATIVE_ENGINE_PATH`, `DD_INTERNAL_TRACE_NATIVE_ENGINE_PATH`, `DD_INTERNAL_CUSTOM_CLR_PROFILER_PATH` in e76521f